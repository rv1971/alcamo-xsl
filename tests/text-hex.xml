<?xml version="1.0" encoding="UTF-8"?>

<?xml-stylesheet href="#xslt" type="text/xml"?>

<a:doc xmlns:a="tag:rv1971@web.de,2021:alcamo-xsl#">
  <xsl:stylesheet
      xmlns:xsl="http://www.w3.org/1999/XSL/Transform"
      xml:id="xslt"
      version="1.0">
    <xsl:import href="../xsl/text.xsl"/>

    <xsl:template match="/">
      <a:result>
        <xsl:apply-templates select="/a:doc/a:data"/>
      </a:result>
    </xsl:template>

    <xsl:template match="a:item">
      <a:item>
        <xsl:value-of select="."/>
        <xsl:text> </xsl:text>
        <xsl:apply-templates select="." mode="a:hex"/>
        <xsl:text> </xsl:text>

        <xsl:apply-templates select="." mode="a:fixed-length-hex">
          <xsl:with-param name="length" select="2"/>
        </xsl:apply-templates>

        <xsl:text> </xsl:text>

        <xsl:apply-templates select="." mode="a:fixed-length-hex">
          <xsl:with-param name="length" select="4"/>
        </xsl:apply-templates>
      </a:item>
    </xsl:template>
  </xsl:stylesheet>

  <a:data>
    <a:item>0</a:item>
    <a:item>1</a:item>
    <a:item>09</a:item>
    <a:item>10</a:item>
    <a:item>11</a:item>
    <a:item>12</a:item>
    <a:item>13</a:item>
    <a:item>14</a:item>
    <a:item>15</a:item>
    <a:item>16</a:item>
    <a:item>17</a:item>
    <a:item>18</a:item>
    <a:item>31</a:item>
    <a:item>32</a:item>
    <a:item>33</a:item>
    <a:item>000255</a:item>
    <a:item>256</a:item>
    <a:item>257</a:item>
    <a:item>65535</a:item>
    <a:item>65536</a:item>
    <a:item>4294967295</a:item>
    <a:item>4294967296</a:item>
    <a:item>4294967297</a:item>
  </a:data>
</a:doc>
