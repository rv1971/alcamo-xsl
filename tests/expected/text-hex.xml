<?xml version="1.0"?>
<a:result xmlns:a="tag:rv1971@web.de,2021:alcamo-xsl#">
  <a:item>0 0 00 0000</a:item>
  <a:item>1 1 01 0001</a:item>
  <a:item>09 9 09 0009</a:item>
  <a:item>10 A 0A 000A</a:item>
  <a:item>11 B 0B 000B</a:item>
  <a:item>12 C 0C 000C</a:item>
  <a:item>13 D 0D 000D</a:item>
  <a:item>14 E 0E 000E</a:item>
  <a:item>15 F 0F 000F</a:item>
  <a:item>16 10 10 0010</a:item>
  <a:item>17 11 11 0011</a:item>
  <a:item>18 12 12 0012</a:item>
  <a:item>31 1F 1F 001F</a:item>
  <a:item>32 20 20 0020</a:item>
  <a:item>33 21 21 0021</a:item>
  <a:item>000255 FF FF 00FF</a:item>
  <a:item>256 100 100 0100</a:item>
  <a:item>257 101 101 0101</a:item>
  <a:item>65535 FFFF FFFF FFFF</a:item>
  <a:item>65536 10000 10000 10000</a:item>
  <a:item>4294967295 FFFFFFFF FFFFFFFF FFFFFFFF</a:item>
  <a:item>4294967296 100000000 100000000 100000000</a:item>
  <a:item>4294967297 100000001 100000001 100000001</a:item>
</a:result>
